Import LC classification from MARC record if existing edition found

<!-- A clear and concise description of what the problem is. Ex. I'm always frustrated when [...] -->

Currently when we import a book from archive.org MARC records, if an existing match is found, the LC classifications (MARC `050`) are not added to the OL record.




### Describe the problem that you'd like solved
<!-- A clear and concise description of what you want to happen. -->

### Proposal & Constraints
<!-- What is the proposed solution / implementation? Is there a precedent of this approach succeeding elsewhere? -->

<!-- Which suggestions or requirements should be considered for how feature needs to appear or be implemented? -->

### Additional context
<!-- Add any other context or screenshots about the feature request here. -->

### Stakeholders
<!-- @ tag stakeholders of this bug -->

While you are looking at this can you please explain the source of the `marc` dict in these locations?
```
./openlibrary/catalog/works/by_author.py:81:24: F821 undefined name 'marc'
    if e['key'] not in marc:
                       ^
./openlibrary/catalog/works/by_author.py:85:12: F821 undefined name 'marc'
    data = marc[e['key']][1]
           ^
./openlibrary/catalog/people/build_object.py:49:29: F821 undefined name 'marc'
    p = build_person_object(marc, [])
                            ^
./openlibrary/catalog/people/build_object.py:53:29: F821 undefined name 'marc'
    p = build_person_object(marc, [])
                            ^
```

--- a/openlibrary/catalog/add_book/__init__.py
+++ b/openlibrary/catalog/add_book/__init__.py
@@ -742,13 +742,11 @@ def load(rec, account=None):
         e['ocaid'] = rec['ocaid']
         need_edition_save = True
 
+    # Add list fields to edition as needed
     edition_fields = [
-        'local_id', 'ia_box_id', 'ia_loaded_id', 'source_records']
-    # TODO:
-    # only consider `source_records` for newly created work
-    # or if field originally missing:
-    #if work_created and not e.get('source_records'):
-    #    edition_fields.append('source_records')
+        'local_id',
+        'source_records',
+        ]
     for f in edition_fields:
         if f not in rec:
             continue

--- a/openlibrary/catalog/add_book/__init__.py
+++ b/openlibrary/catalog/add_book/__init__.py
@@ -745,6 +745,8 @@ def load(rec, account=None):
     # Add list fields to edition as needed
     edition_fields = [
         'local_id',
+        'lccn',
+        'lc_classifications',
         'source_records',
         ]
     for f in edition_fields:

